### PR TITLE
Make mxnet build successful in CPU

### PR DESCRIPTION
**Pull request by apeforest**
Two changes:
1) remove the CBLAS flag for MSHASOW.
2) enclose code that works only in GPU with HAVE_CUDA env variable.

After this change, user can simply run the following to install MXNet-Horovod on CPU instances (I tested on my macbook):

`pip install --upgrade -v --no-cache-dir  . `

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -514,7 +514,12 @@ def get_common_options(build_ext):
         LIBRARIES += ['ddl', 'ddl_pack']
 
     if mxnet_include_dirs:
-        MACROS += [('MSHADOW_USE_CBLAS', '1')]
+        #MACROS += [('MSHADOW_USE_CBLAS', '1')]
```
Comment (yuxihu) [inline setup.py]: remove this commented line

File: setup.py [hunk 2]
```diff
@@ -514,7 +514,12 @@ def get_common_options(build_ext):
...
+        if have_cuda:
+            MACROS += [('MSHADOW_USE_CUDA', '1')]
+        else:
+            MACROS += [('MSHADOW_USE_CUDA', '0')] 
+        MACROS += [('MSHADOW_USE_MKL', '0')]
```
Comment (yuxihu) [inline setup.py]: why set MSHADOW_USE_MKL to 0 here?

File: horovod/mxnet/mpi_ops.cc [hunk 3]
```diff
@@ -202,68 +202,58 @@ extern "C" int horovod_mxnet_allreduce_async(
       RunContext rctx, Engine::CallbackOnComplete cb) mutable {
       DoAllreduce(input, output, average, new_name, cb);
   };
+#if HAVE_CUDA
```
Comment (yuxihu) [inline horovod/mxnet/mpi_ops.cc]: The way that github shows the difference is hard to follow. Let's discuss offline about these changes tomorrow.

File: horovod/mxnet/mpi_ops.cc [hunk 4]
```diff
@@ -202,68 +202,58 @@ extern "C" int horovod_mxnet_allreduce_async(
...
   auto allreduce_async_cpu_fn = [input, output, new_name, average](
       RunContext rctx, Engine::CallbackOnComplete cb) mutable {
       DoAllreduceCudaOnCPU(input, output, average, new_name, cb);
   };
-  int cpu = -1;
-#ifdef HOROVOD_GPU_ALLREDUCE
-  if (input->ctx().dev_mask() == gpu::kDevMask &&
```
Comment (ctcyang) [inline horovod/mxnet/mpi_ops.cc]: Does the removal of this if-statement cause any problems? Was it a bug to include the if-statement here in the first place?

File: setup.py (hunk 2 above)
Comment (apeforest) [reply setup.py]: Because my MacBook does not have MKL library and failed to build. I will add a function to check the existence of MKL package here.

File: horovod/mxnet/mpi_ops.cc (hunk 3 above)
Comment (apeforest) [reply horovod/mxnet/mpi_ops.cc]: Discussed offline.

File: horovod/mxnet/mpi_ops.cc (hunk 4 above)
Comment (apeforest) [reply horovod/mxnet/mpi_ops.cc]: This piece of code puzzles me as well. It seems the purpose is not for functionality by performance concern? Did you add this check here? I don't find similar checking in PyTorch and Tensorflow, thus wondering if this is indeed needed.

Comment (ctcyang) [reply horovod/mxnet/mpi_ops.cc]: Yes, I added it. I just remembered this is a layover of when we were still using CUDA-aware MPI. My reasoning back then was I wanted it to skip the copy-to-CPU when both input and output were on GPU. Since we are not using CUDA-aware MPI anymore, we can get rid of it.
